### DISCUSSION TITLE
Filling form with multiple "set" of fields of the same name

### DISCUSSION THREAD

**Discussion by Trigve** [py-pdf/pypdf#1401] [Q&A]
As first I'm not a PDF expert and don't know all the terminology.

I've a PDF with 2 columns of the fields. Those fields have the same name but are somehow differentiated by the "set". That is, I've a sample FDF for the PDF in which when I want to fill values for the 1. column I need to use something like
`<< /T (1)/Kids
	  [ ... ]
`
On the other hand if I want to fill the 2. column I need to use something like
`<< /T (2)/Kids
	  [ ... ]
`
Now, when usin PyPDF2 `update_page_form_field_values()` both column's fields ende having same values. Could I somehow set which part of the form values I want to update (even hacks for now)?

I could post the .pdf if needed.

**Comment by Trigve** [comment]
So after digging a little bit I found that PyPDF2 doesn't use fully qualified named for hierarchical form fields. That is in my case, there is "1.1.some_field" but PyPDF2 would return "some_field" only.

I've implemented the fully qualified name support with optional method arguments so original behaviour is preserved. I could make a pull request if there is some interest for this feature.

**Reply by pubpub-zz** [reply]
Any improvement to PyPDF2 is always welcomed 😊
Looking forward to see that. I'm currently finishing some improvement on the merging/appending of documents : You may be interested in having a look also on how to add the hierarchical structure in this function.

**Reply by MartinThoma** [reply]
I agree - improvements are always welcome! I am aware that many issues exist with forms and how PyPDF2 deals with them. I would be happy if we could resolve them.

**Reply by Trigve** [reply]
@pubpub-zz I can look at it.

I will need to clean up the code and will create the PR then.

**Reply by alexeysergeev-cm** [reply]
hey, is there a solution to get a full field name not partial?
